src/modules/deal-folder/deal-folder.service.ts

```diff
@@ -0,0 +1,175 @@
+    fileName: string,
+    dealId: string,
+    buyerName: string,
+    ukefSiteId: string,
+    documentType: DocumentTypeEnum,
+  ): Promise<string> {
+    const { urlToUpdateFileInfo, fileLeafRef } = await this.constructUrlToUpdateFileInfo(fileName, dealId, buyerName, ukefSiteId);
+    const requestBodyToUpdateFileInfo = this.constructRequestBodyToUpdateFileInfo(documentType);
+
+    await this.graphService.patch<
+      {
+        Title: string;
+        Document_x0020_Status: string;
+      },
+      unknown
+    >({
+      path: urlToUpdateFileInfo,
+      requestBody: requestBodyToUpdateFileInfo,
+    });
+
+    return fileLeafRef;
+  }
+
+  private async constructUrlToCreateUploadSession(fileName: string, dealId: string, buyerName: string, ukefSiteId: string): Promise<string> {
+    const sharepointSiteId = await this.getSharepointSiteIdByUkefSiteId(ukefSiteId);
+    const driveId = await this.getResourceIdByName(ukefSiteId, CASE_LIBRARY.DRIVE_NAME, ENUMS.SHAREPOINT_RESOURCE_TYPES.DRIVE);
+    // Getting the sharepointSiteId () and driveId appears to necessary because using the ukefSiteId and drive name only appears to work when accessing drives,
+    // not items *within* drives.
+    const fileDestinationPath = `${buyerName}/D${dealId}`;
+
+    return `https://graph.microsoft.com/v1.0/sites/${sharepointSiteId}/drives/${driveId}/root:/${fileDestinationPath}/${fileName}:/createUploadSession`;
+  }
+
+  private getSharepointSiteIdByUkefSiteId(ukefSiteId: string): Promise<string> {
+    return this.graphService
+      .get<{ id: string }>({
+        path: `sites/${this.config.ukefSharepointName}:/sites/${ukefSiteId}`,
+      })
+      .then((site) => site.id);
+  }
+
+  private getResourceIdByName(ukefSiteId: string, resourceName: string, sharepointResourceType: SharepointResourceTypeEnum): Promise<string> {
+    return this.graphService
+      .get<{ value: { name: string; id: string }[] }>({
+        path: `sites/${this.config.ukefSharepointName}:/sites/${ukefSiteId}:/${sharepointResourceType}s`,
+      })
+      .then((response) => response.value)
+      .then((resources) => resources.find((resource) => resource.name === resourceName)) // This is necessary because the 'filter' query parameter does not currently support the 'name' field on lists/list items (see comment from Microsoft employee https://learn.microsoft.com/en-us/answers/questions/980144/graph-api-sharepoint-list-filter-by-createddatetim).
+      .then((resource) => resource.id); // Additionally, the 'filter' query parameter is not supported at all by the 'List available drives' endpoint (see https://learn.microsoft.com/en-us/graph/api/drive-list?view=graph-rest-1.0&tabs=http#optional-query-parameters for a list of supported parameters).
+  }
+
+  private constructRequestBodyToUpdateFileInfo(documentType: string): {
+    Title: string;
+    Document_x0020_Status: string;
+  } {
+    const { documentTitle, documentTypeId } = DOCUMENT_TYPE_MAPPING[documentType];
```

Comment:
We've got warning here
> Variable Assigned to Object Injection Sink
  
I think introducing a mapping class as suggested above might fix that too

Reply:
I used a switch statement in the method on the mapping class which fixed this